src/context/WalletProvider/WalletConnect/components/Connect.tsx

```diff
@@ -0,0 +1,96 @@
+import { WalletConnectHDWallet } from '@shapeshiftoss/hdwallet-walletconnect'
+import { getConfig } from 'config'
+import React, { useState } from 'react'
+import { RouteComponentProps } from 'react-router-dom'
+import { ActionTypes, WalletActions } from 'context/WalletProvider/actions'
+import { KeyManager } from 'context/WalletProvider/KeyManager'
+import { setLocalWalletTypeAndDeviceId } from 'context/WalletProvider/local-wallet'
+import { useWallet } from 'hooks/useWallet/useWallet'
+
+import { ConnectModal } from '../../components/ConnectModal'
+import { LocationState } from '../../NativeWallet/types'
+import { WalletConnectConfig } from '../config'
+
+export interface WalletConnectSetupProps
+  extends RouteComponentProps<
+    {},
+    any, // history
+    LocationState
+  > {
+  dispatch: React.Dispatch<ActionTypes>
+}
+
+/**
+ * WalletConnect Connect component
+ *
+ * Test WalletConnect Tool: https://test.walletconnect.org/
+ */
+export const WalletConnectConnect = ({ history }: WalletConnectSetupProps) => {
+  const { dispatch, state } = useWallet()
+  const [loading, setLoading] = useState(false)
+  const [error, setError] = useState<string | null>(null)
+
+  const setErrorLoading = (e: string | null) => {
+    setError(e)
+    setLoading(false)
+  }
+
+  const pairDevice = async () => {
+    setError(null)
+    setLoading(true)
+
+    const rpcUrl = getConfig().REACT_APP_ETHEREUM_NODE_URL
+    if (!rpcUrl) {
```

Comment:
I'm wondering if this case is ever hit: Tried adding a debugger before that block, and if `REACT_APP_ETHEREUM_NODE_URL` isn't present in config, `EnvMissingError: REACT_APP_ETHEREUM_NODE_URL` is thrown by `getConfig()`, and the execution stops there for `pairDevice()`.

This is probably a logic coming over from previous `process.env.X_Y_Z` checks, where indeed, an undefined env var would result in an `undefined` expression, whereas using `getconfig` results in a thrown error.

Reply:
Looking at this now, I agree - this code isn't reachable. There is no such check in `metamask/Connect.tsx` either, so I'll remove these lines here.